libs/linking-tool/README.md

```diff
@@ -1,12 +1,203 @@
 # linking-tool
 
-This library was generated with [Nx](https://nx.dev).
+Library for managing routes in Angular application.
 
-## Running unit tests
+There are 2 ways of handling routes - you can create a tree of links, or a map. Both are plain JSON text objects.
 
-Run `nx test linking-tool` to execute the unit tests.
+Table of contents
+-
+1. Generating links
+    - using a map
+    - using a tree
+2. Using links service in an app
+    - with a links map
+    - with a links tree
 
+Generating links
+-
+Common requirements:
+- every routing module name has to fit the following pattern `*-routing.module.ts`
+- route declarations have to be in a `routes` local variable
 
-## Usage
+Common recommendations:
+- create some enum with link types, like the following:
+```
+export enum LinkType {
+    Admin = 'admin',
```

Comment:
I'm not sure it will work with current implementation. It seems like enum item name and its value must match exactly 1:1, because I think you use item name instead of value when saving items to map/tree.

Reply:
Good point, I will fix it.